src/diagnostics.jl

```diff
@@ -149,6 +149,11 @@ end
     P.profile /= (Nx*Ny)  # Normalize to get the mean from the sum.
 end
 
+function (p::HorizontalAverage)(model)
+    run_diagnostic(model, p)
+    return p.profile

```

Comment:
Hmm. This cannot be used as-is because `p.profile` is a `CuArray`, correct?

Reply:
I think it's fine but there's a design choice to be made:

Does the `HorizontalAverage` (and diagnostics in general)
1. always return an `Array` (which can be surprising if you're running a GPU model, and you may want it to be a CuArray), or
2. should it return a `CuArray` when running a GPU model (and it's up to the caller to do the conversion is needed)?

I would argue the second choice is more consistent and flexible.

Reply:
Hmm. I sympathize with your concern about surprisingness and flexibility --- though I'm not quite sure when it would be useful to actually return a `CuArray` (this doesn't mean such a case may not someday arise...)

I don't quite understand how output would work with option 2. Can you give an example? Would the user have to write a function that references a `HorizontalAverage` instances as global?

There's a third option, which is to add a field to `HorizontalAverage` (something like `ReturnType :: RT`) Then the function you've written can dispatch on `RT <: Nothing`. Otherwise, we'd write `return ReturnType(p.profile)`.

The advantage of this third option is 1) it avoids the aforementioned concerns and 2) it permits pattens like

```julia
Tavg = HorizontalAverage(model.tracers.T, return_type=Array{Float32})
output = JLD2OutputWriter(model, (Tavg=Tavg,), ...)
```

etc.